queries/nix.scm

```diff
@@ -0,0 +1,146 @@
+;; Statements
+
+[
+    (binding)
+] @statement
+;; This matches assert statements as stand-alone statements. This
+;; is because it's common to have a series of assert statements in a file, which
+;; would otherwise all match as a single statement. See the playground/nix/asserts.nix
+;; file for an example.
+(assert_expression
+    "assert" @statement.start
+    condition: (_)
+    ";" @statement.end
+    body: (_)
+)
+(binding_set) @map.iteration @list.iteration @ifStatement.iteration
+(source_code) @statement.iteration @map.iteration @comment.iteration
+
+;;!! let
+;;!!   a = 1;
+;;!    <*****
+;;!!   b = 2;
+;;!    *****>
+;;!! in a + b
+(let_expression
+    "let"
+    (_) @statement.iteration @value.iteration @name.iteration
+)
+
+;; Conditionals
+
+;;!! key = if a then b else c;
+;;!        ^^^^^^^^^^^^^^^^^^
+(if_expression) @ifStatement @branch.iteration
+;;!! key = if a then b else c;
+;;!        ^^^^
+(if_expression
+    "if" @branch.start
+    condition: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!             ^^^^^^
+(if_expression
+    "then" @branch.start
+    consequence: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!                    ^^^^^^
+(if_expression
+    "else" @branch.start
+    alternative: (_) @branch.end
+)
+
+;; Lists and maps
+
+;;!! foo = [ a b c ];
+;;!        ^^^^^^^^^
+(list_expression) @list
+
+;;!! foo = { x = 1; y = 2; };
+;;!        ^^^^^^^^^^^^^^^^^^
+[
+    (attrset_expression)
+    (rec_attrset_expression)
+] @map @statement.iteration @value.iteration @name.iteration
+
+;;!! foo = { x = 1; y = 2; };
+;;!          ^
+(attrpath) @collectionKey
+
+;; Strings
+
+;;!! # foo
+;;!  ^^^^^
+(comment) @comment @textFragment
+
+(string_expression
+    _ @textFragment.start.endOf
+    (string_fragment)
+    _ @textFragment.end.startOf
+) @string
+
+(indented_string_expression
+    "''" @textFragment.start.endOf
+    (string_fragment)
+    "''" @textFragment.end.startOf
+) @string
+
+;; Hack to test these two are otherwise sane selectors
+;; [
+;;     (string_expression)
+;;     (indented_string_expression)
+;; ] @comment
+
+;; Functions
+
+;; Note for this part of the function, we identify is as lambda only
+;;!! x = a: b: a + b;
+;;!         ^^^^^^^^ Func1 due to currying
+;;!      ^^^^^^^^^^^ Func2 due to currying
+(function_expression
+    universal: (identifier) @argumentOrParameter
+    body: (_)
+) @anonymousFunction @argumentOrParameter.iteration
+
+;; We define the named function as the full assignment of the lambda
+;; This means funk name becomes the variable holding the lambda
+(binding
+    (_) @functionName
+    "="
+    expression: (function_expression) @namedFunction.interior
```

Comment:
if you're supporting `interior`, it's worth having a `"change inside funk"` test

Reply:
Added it, albeit it not sure how useful `@namedFunction.interior` actually is in the nix case, but I think it doesn't hurt... basically the same as `change value`. Not sure how you folks feel about that type of redundancy. I guess it may depend on whether or not ultimately others agree my calling the assignment of a lambda to a variable a named function is sane. If not I guess I may drop the named function stuff altogether.

Reply:
I added a `change inside funk` and a `change inside lambda` test